### PR TITLE
DAOS-11239 control: Stub SPDK library interface in go unit tests

### PR DESCRIPTION
**Pull request by tanabarr**
Provide stubbing for the SPDK C library interface so that go does not
link to the actual library unless the spdk build flag is provided.

Also tidy stubbing of the libipmctl C library interface.

Required-githooks: true

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/control/lib/spdk/nvme.go [hunk 1]
```diff
@@ -6,39 +6,22 @@
...
 	"os"
-	"strings"
-	"unsafe"
-
-	"github.com/pkg/errors"
 
 	"github.com/daos-stack/daos/src/control/common"
 	"github.com/daos-stack/daos/src/control/logging"
 	"github.com/daos-stack/daos/src/control/server/storage"
+	"github.com/pkg/errors"
```
Comment (mjmac) [inline src/control/lib/spdk/nvme.go:15]: nit: misplaced import

**Review by mjmac — COMMENTED** [review]
Seems like a good cleanup.

File: src/control/lib/spdk/nvme.go (hunk 1 above)
Comment (tanabarr) [reply src/control/lib/spdk/nvme.go]: done

**Review by jolivier23 — APPROVED** [review]

File: src/control/lib/ipmctl/nvm_ctest_linux_amd64.go [hunk 2]
```diff
@@ -1,5 +1,5 @@
 //
-// (C) Copyright 2018-2022 Intel Corporation.
+// (C) Copyright 2022 Intel Corporation.
```
Comment (kjacque) [inline src/control/lib/ipmctl/nvm_ctest_linux_amd64.go:2]: Looks like some unbidden copyright changes?

File: src/control/lib/ipmctl/nvm_stubs.go [hunk 3]
```diff
@@ -1,5 +1,5 @@
 //
-// (C) Copyright 2018-2022 Intel Corporation.
+// (C) Copyright 2022 Intel Corporation.
```
Comment (kjacque) [inline src/control/lib/ipmctl/nvm_stubs.go:2]: The old copyright range is still correct

**Review by kjacque — APPROVED** [review]
Just some copyright nits. OK landing without fixing.

Is the overall goal that we don't build spdk for unit tests?

File: src/control/lib/ipmctl/nvm_ctest_linux_amd64.go (hunk 2 above)
Comment (tanabarr) [reply src/control/lib/ipmctl/nvm_ctest_linux_amd64.go:2]: but the file didn't exist until 2022

File: src/control/lib/ipmctl/nvm_stubs.go (hunk 3 above)
Comment (tanabarr) [reply src/control/lib/ipmctl/nvm_stubs.go:2]: as per previous comment

**Pull request merged**